src/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/Tov.cpp

```diff
@@ -0,0 +1,160 @@
+  std::vector<double> log_enthalpy;
+};
+
+}  // namespace
+
+namespace EinsteinSolutions {
+
+TovSolution::TovSolution(
+    const std::unique_ptr<EquationsOfState::EquationOfState<true, 1>>&
+        equation_of_state,
+    const double central_mass_density, const double final_log_enthalpy,
+    const double absolute_tolerance, const double relative_tolerance) {
+  std::array<double, 2> u_and_v = {{0.0, 0.0}};
+  std::array<double, 2> dudh_and_dvdh{};
+  const double central_log_enthalpy =
+      std::log(get(equation_of_state->specific_enthalpy_from_density(
+          Scalar<double>{central_mass_density})));
+  lindblom_rhs(&dudh_and_dvdh, u_and_v, central_log_enthalpy,
+               equation_of_state);
+  const double initial_step = -std::min(std::abs(1.0 / dudh_and_dvdh[0]),
+                                        std::abs(1.0 / dudh_and_dvdh[1]));
+  using StateDopri5 =
+      boost::numeric::odeint::runge_kutta_dopri5<std::array<double, 2>>;
+  boost::numeric::odeint::dense_output_runge_kutta<
+      boost::numeric::odeint::controlled_runge_kutta<StateDopri5>>
+      Dopri5 = make_dense_output(absolute_tolerance, relative_tolerance,
+                                 StateDopri5{});
+  Observer observer{};
+  boost::numeric::odeint::integrate_adaptive(
+      Dopri5,
+      [&equation_of_state](const std::array<double, 2>& lindblom_u_and_v,
+                           std::array<double, 2>& lindblom_dudh_and_dvdh,
+                           const double lindblom_enthalpy) noexcept {
+        return lindblom_rhs(&lindblom_dudh_and_dvdh, lindblom_u_and_v,
+                            lindblom_enthalpy, equation_of_state);
+      },
+      u_and_v, central_log_enthalpy, final_log_enthalpy, initial_step,
+      std::ref(observer));
+  outer_radius_ = observer.radius.back();
+  mass_interpolant_ =
+      intrp::BarycentricRational(observer.radius, observer.mass, 5);
+  // log_enthalpy(radius) is almost linear so an interpolant of order 3
+  // maximizes precision
+  log_enthalpy_interpolant_ =
+      intrp::BarycentricRational(observer.radius, observer.log_enthalpy, 3);
+}
+
+double TovSolution::outer_radius() const noexcept { return outer_radius_; }
+double TovSolution::mass(const double r) const noexcept {
+  return mass_interpolant_(r);
+}
+Scalar<DataVector> TovSolution::mass(const Scalar<DataVector>& radius) const
+    noexcept {
+  DataVector mass(radius.size(), 0.0);
+  for (size_t i = 0; i < radius.size(); i++) {
+    mass[i] = mass_interpolant_(get(radius)[i]);
+  }
+  return Scalar<DataVector>{mass};
+}
+double TovSolution::specific_enthalpy(const double r) const noexcept {
+  return std::exp(log_enthalpy_interpolant_(r));
+}
+Scalar<DataVector> TovSolution::specific_enthalpy(
+    const Scalar<DataVector>& radius) const noexcept {
+  DataVector specific_enthalpy(radius.size(), 0.0);
+  for (size_t i = 0; i < radius.size(); i++) {
+    specific_enthalpy[i] = log_enthalpy_interpolant_(get(radius)[i]);
+  }
```

Comment:
Use the `log_specific_enthalpy` method that does this and just exponentiate the result.